### PR TITLE
Bigtable system tests passing against V2 endpoints

### PR DESCRIPTION
**Pull request by tseaver**
/cc @garye, @lesv, @sduskis, @tswast, @mbrukman, @jgeewax

### CODE REVIEW

File: gcloud/bigtable/client.py [hunk 1]
```diff
@@ -375,22 +376,33 @@ def __exit__(self, exc_type, exc_val, exc_t):
...
+    def instance(self, instance_id, location,
+                 display_name=None, serve_nodes=DEFAULT_SERVE_NODES):
         """Factory to create a instance associated with this client.
 
         :type instance_id: str
         :param instance_id: The ID of the instance.
 
+        :type location: string
+        :param location: location name, in form
+                         ``projects/<project>/locations/<location>``; used to
```
Comment (garye) [inline gcloud/bigtable/client.py]: Doesn't the client already know the project? Could the bare location be the only argument here?

File: system_tests/bigtable.py [hunk 2]
```diff
@@ -89,114 +84,119 @@ def _operation_wait(operation, max_attempts=5):
     return True
 
 
+def _retry_backoof(meth, *args, **kw):
```
Comment (garye) [inline system_tests/bigtable.py]: backoof -> backoff

File: gcloud/bigtable/client.py (hunk 1 above)
Comment (tseaver) [inline gcloud/bigtable/client.py]: @garye Good point.

File: gcloud/bigtable/instance.py [hunk 3]
```diff
@@ -199,17 +212,28 @@ class Instance(object):
     :param client: The client that owns the instance. Provides
                    authorization and a project ID.
 
+    :type location: string
+    :param location: location name, in form
```
Comment (mbrukman) [inline gcloud/bigtable/instance.py]: location name of the form

File: gcloud/bigtable/client.py [hunk 4]
```diff
@@ -375,22 +376,33 @@ def __exit__(self, exc_type, exc_val, exc_t):
...
-    def instance(self, instance_id, display_name=None):
+    def instance(self, instance_id, location,
+                 display_name=None, serve_nodes=DEFAULT_SERVE_NODES):
         """Factory to create a instance associated with this client.
 
         :type instance_id: str
         :param instance_id: The ID of the instance.
 
+        :type location: string
+        :param location: location name, in form
```
Comment (mbrukman) [inline gcloud/bigtable/client.py]: location name of the form

File: gcloud/bigtable/client.py (hunk 1 above)
Comment (tseaver) [inline gcloud/bigtable/client.py]: d8a01c2

File: system_tests/bigtable.py (hunk 2 above)
Comment (tseaver) [inline system_tests/bigtable.py]: 893d451

File: gcloud/bigtable/instance.py (hunk 3 above)
Comment (tseaver) [inline gcloud/bigtable/instance.py]: @mbrukman  Thanks we are replacing it with `location_id` everywhere.

**Comment by tswast** [comment]
@tseaver Once this goes in, can you start a PyPi release? Then I will be able to update the code samples to use this version.

**Comment by tseaver** [comment]
@tswast This PR is still targeting the 'bigtable-v2' feature branch.  I also need to get the `gcloud.bigtable.happybase` code working, including its separate system tests, before merging the feature branch to `master` and making a release.

**Comment by tswast** [comment]
I get an error when I try to use the code from this branch.

```
$ python main.py swast-bigtable-examples my-cluster us-central1-c
D0628 14:22:01.934576886    6633 ev_posix.c:101]             Using polling engine: poll
Traceback (most recent call last):
  File "main.py", line 29, in <module>
    from gcloud import bigtable
  File "/usr/local/google/home/swast/src/gcloud-python/gcloud/bigtable/__init__.py", line 18, in <module>
    from gcloud.bigtable.client import Client
  File "/usr/local/google/home/swast/src/gcloud-python/gcloud/bigtable/client.py", line 34, in <module>
    from gcloud.bigtable._generated_v2 import (
  File "/usr/local/google/home/swast/src/gcloud-python/gcloud/bigtable/_generated_v2/bigtable_instance_admin_pb2.py", line 6, in <module>
    from google.protobuf import descriptor as _descriptor
ImportError: cannot import name descriptor
```

**Comment by garye** [comment]
Do you need a more recent version of google.protobuf?

**Comment by tswast** [comment]
@garye This is from a fresh virtualenv. I installed via `python setup.py develop`.

```
$ pip freeze | grep protobuf
protobuf==3.0.0b3
```

Version 3.0.0b3 appears to be the [latest release](https://pypi.python.org/pypi/protobuf).

**Comment by tseaver** [comment]
@tswast I cannot reproduce with the current HEAD of the `bigtable-v2` branch:

```bash
$ /opt/Python-2.7.11/bin/virtualenv /tmp/issue_1930
$ cd /tmp/issue_1930
$ mkdir src; cd src
$ git clone -b bigtable-v2 --depth 1 git@github.com:GoogleCloudPlatform/gcloud-python
Cloning into 'gcloud-python'...
remote: Counting objects: 409, done.
remote: Compressing objects: 100% (384/384), done.
remote: Total 409 (delta 77), reused 103 (delta 21), pack-reused 0
Receiving objects: 100% (409/409), 5.76 MiB | 392.00 KiB/s, done.
Resolving deltas: 100% (77/77), done.
Checking connectivity... done.
$ cd ..
$ bin/pip install -e src/gcloud-python/
Obtaining file:///tmp/tswast/src/gcloud-python
Collecting httplib2>=0.9.1 (from gcloud==0.16.0)
Collecting googleapis-common-protos (from gcloud==0.16.0)
Collecting oauth2client>=2.0.1 (from gcloud==0.16.0)
Collecting protobuf!=3.0.0.b2.post1,>=3.0.0b2 (from gcloud==0.16.0)
  Using cached protobuf-3.0.0b3-py2-none-any.whl
Collecting six (from gcloud==0.16.0)
  Using cached six-1.10.0-py2.py3-none-any.whl
Collecting grpcio>=0.14.0 (from gcloud==0.16.0)
  Using cached grpcio-0.15.0-cp27-cp27m-manylinux1_x86_64.whl
Collecting google-gax>=0.12.0 (from gcloud==0.16.0)
Collecting gax-google-pubsub-v1 (from gcloud==0.16.0)
Collecting rsa>=3.1.4 (from oauth2client>=2.0.1->gcloud==0.16.0)
  Using cached rsa-3.4.2-py2.py3-none-any.whl
Collecting pyasn1-modules>=0.0.5 (from oauth2client>=2.0.1->gcloud==0.16.0)
  Using cached pyasn1_modules-0.0.8-py2.py3-none-any.whl
Collecting pyasn1>=0.1.7 (from oauth2client>=2.0.1->gcloud==0.16.0)
  Using cached pyasn1-0.1.9-py2.py3-none-any.whl
Requirement already satisfied (use --upgrade to upgrade): setuptools in ./lib/python2.7/site-packages (from protobuf!=3.0.0.b2.post1,>=3.0.0b2->gcloud==0.16.0)
Collecting enum34>=1.0.4 (from grpcio>=0.14.0->gcloud==0.16.0)
  Using cached enum34-1.1.6-py2-none-any.whl
Collecting futures>=2.2.0 (from grpcio>=0.14.0->gcloud==0.16.0)
  Using cached futures-3.0.5-py2-none-any.whl
Collecting ply==3.8 (from google-gax>=0.12.0->gcloud==0.16.0)
Collecting grpc-google-pubsub-v1>=0.7.10 (from gax-google-pubsub-v1->gcloud==0.16.0)
Installing collected packages: httplib2, six, protobuf, googleapis-common-protos, pyasn1, rsa, pyasn1-modules, oauth2client, enum34, futures, grpcio, ply, google-gax, grpc-google-pubsub-v1, gax-google-pubsub-v1, gcloud
  Running setup.py develop for gcloud
Successfully installed enum34-1.1.6 futures-3.0.5 gax-google-pubsub-v1-0.7.10 gcloud google-gax-0.12.1 googleapis-common-protos-1.1.0 grpc-google-pubsub-v1-0.7.10 grpcio-0.15.0 httplib2-0.9.2 oauth2client-2.2.0 ply-3.8 protobuf-3.0.0b3 pyasn1-0.1.9 pyasn1-modules-0.0.8 rsa-3.4.2 six-1.10.0
$ bin/python
Python 2.7.11 (default, Feb 22 2016, 07:44:44) 
[GCC 4.8.4] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> from gcloud import bigtable
D0629 16:41:43.553773202   30989 ev_posix.c:101]             Using polling engine: poll
>>>
```

**Pull request merged**